install/kubernetes/helm/istio/files/injection-template.yaml

```diff
@@ -27,36 +29,52 @@ initContainers:
   - "{{ annotation .ObjectMeta `traffic.sidecar.istio.io/includeInboundPorts` `*` }}"
   - "-d"
   - "{{ excludeInboundPort (annotation .ObjectMeta `status.sidecar.istio.io/port` .Values.global.proxy.statusPort) (annotation .ObjectMeta `traffic.sidecar.istio.io/excludeInboundPorts` .Values.global.proxy.excludeInboundPorts) }}"
-  {{ if or (isset .ObjectMeta.Annotations `traffic.sidecar.istio.io/excludeOutboundPorts`) (ne .Values.global.proxy.excludeOutboundPorts "") -}}
+  {{ if or (isset .ObjectMeta.Annotations `traffic.sidecar.istio.io/excludeOutboundPorts`) (ne (valueOrDefault .Values.global.proxy.excludeOutboundPorts "") "") -}}
   - "-o"
   - "{{ annotation .ObjectMeta `traffic.sidecar.istio.io/excludeOutboundPorts` .Values.global.proxy.excludeOutboundPorts }}"
   {{ end -}}
   {{ if (isset .ObjectMeta.Annotations `traffic.sidecar.istio.io/kubevirtInterfaces`) -}}
   - "-k"
   - "{{ index .ObjectMeta.Annotations `traffic.sidecar.istio.io/kubevirtInterfaces` }}"
   {{ end -}}
-  imagePullPolicy: "{{ .Values.global.imagePullPolicy }}"
+  {{ if .Values.istio_cni.enabled -}}
+  - "--run-validation"
+  - "--skip-rule-apply"
+  {{ end -}}
+  imagePullPolicy: "{{ valueOrDefault .Values.global.imagePullPolicy `Always` }}"
 {{- if .Values.global.proxy.init.resources }}
   resources:
     {{ toYaml .Values.global.proxy.init.resources | indent 4 }}
 {{- else }}
   resources: {}
 {{- end }}
   securityContext:
+  {{- if not .Values.istio_cni.enabled }}
     allowPrivilegeEscalation: {{ .Values.global.proxy.privileged }}
+    privileged: {{ .Values.global.proxy.privileged }}
+  {{- else }}
+    allowPrivilegeEscalation: false
+    privileged: false
+  {{- end  }}
     capabilities:
+  {{- if not .Values.istio_cni.enabled }}
       add:
       - NET_ADMIN
       - NET_RAW
+  {{- end }}
       drop:
       - ALL
-    privileged: {{ .Values.global.proxy.privileged }}
     readOnlyRootFilesystem: false
+  {{- if not .Values.istio_cni.enabled }}
     runAsGroup: 0
     runAsNonRoot: false
     runAsUser: 0
+  {{- else }}
+    runAsGroup: 1337
+    runAsUser: 1337
```

Comment:
So this change runs the normal iptables code as a non-root user? That's not related to this PR's issue, but that's nice, thanks!

Reply:
Oh actually, this is wrong. When CNI is enabled, you run the verification code with UID and GID 1337, so redirection will be disabled.
You should invert the logic: we can now run the normal istio-init (non-CNI case) with a UID/GID 1337, but you can use neither UID/GID 0 nor 1337 for the CNI case for redirection detection.

Reply:
Talked offline: the validation is supposed to bypass the earlier rules and hit the 1337 rule below. So that is why validation need to run with uid and gid 1337.

```
iptables -t nat -A OUTPUT -p tcp -j ISTIO_OUTPUT
iptables -t nat -A ISTIO_OUTPUT -o lo -s 127.0.0.6/32 -j RETURN
iptables -t nat -A ISTIO_OUTPUT -o lo ! -d 127.0.0.1/32 -m owner --uid-owner 1337 -j ISTIO_IN_REDIRECT
```

Reply:
LGTM. I was expecting to hit the same `REDIRECT` rule that is hit by connections from application containers, not this rule hit by connections from Envoy. But hitting that other rule would require using a UID/GID different from 1337.
A good point about keeping UID 1337 here is that we can't use UID 0 since it's typically forbidden by PodSecurityPolicies, and any other UID may also be forbidden by PSPs.
So this LGTM.